src/status.ts

```diff
@@ -23,123 +30,124 @@ const STARTED_SERVER_OPTIONS = [
   { label: "Ruby LSP: Restart", description: Command.Restart },
 ];
 
-export class StatusItem {
-  private context: vscode.ExtensionContext;
-  private ruby: Ruby;
-  private selector: vscode.DocumentSelector;
-  private serverStatus: vscode.LanguageStatusItem;
-  private yjitStatus: vscode.LanguageStatusItem;
-  private rubyVersionStatus: vscode.LanguageStatusItem;
-  private experimentalFeaturesStatus: vscode.LanguageStatusItem;
-
-  constructor(context: vscode.ExtensionContext, ruby: Ruby) {
-    this.context = context;
-    this.ruby = ruby;
-    this.selector = {
+export interface Client {
+  context: vscode.ExtensionContext;
+  ruby: Ruby;
+  state: ServerState;
+}
+
+export abstract class StatusItem {
+  public item: vscode.LanguageStatusItem;
+  protected context: vscode.ExtensionContext;
+
+  constructor(id: string, client: Client) {
+    this.item = vscode.languages.createLanguageStatusItem(id, {
       scheme: "file",
       language: "ruby",
+    });
+    this.context = client.context;
+    this.registerCommand();
+  }
+
+  abstract refresh(client: Client): void;
```

Comment:
Do we still need the client here, if we were to store the `client` inside `constructor` and used `this.client` inside the `refresh` method to access properties?

Reply:
We don't indeed. Refresh takes no arguments now